### PR TITLE
Add ability to convert from a hex string to Color32 using macros

### PR DESCRIPTION
**Pull request by newcomb-luke**
This PR closes <https://github.com/emilk/egui/issues/1492>.

Again currently `./sh/check.sh` is currently failing for me, see my other [PR](https://github.com/emilk/egui/pull/1595).
Cargo clippy, cargo test, and cargo fmt have all been run.
The CHANGELOG.md has been updated under "Unreleased"

This change allows you to create Color32's from hex strings:

```rust
let color = epaint::hex_rgb!("#4c4c4c");
```

And it does so at compile-time.

### CODE REVIEW

**Comment by emilk** [comment]
Very nice!

But I would suggest we simplify it. As far as I can recall I've never seen premultiplied alpha expressed in hex form. So I suggest we boil it down to a single macro that accepts both RGB and unmultiplied RGBA (`epaint::color_hex!("#55eedd") == epaint::color_hex!("#55eeddff")`).

For additive, users can just do `epaint::color_hex!("#ffeedd").additive()` (we can make `Color32::additive` `const fn`)

**Comment by newcomb-luke** [comment]
Upon further testing, previously the from_rgb, from_rgb_additive, from_rgba_premultiplied were `const`, however from_rgba_unmultiplied due to the floating-point conversions that have to take place, is unable to be `const` [yet](https://github.com/rust-lang/rust/issues/57241). So only the conversion from the string literal to the array of values can be const. So the macro invocation itself cannot yet be `const`, but soon it could be. This isn't much slower than using the regular from_rgba_unmultiplied though.

File: epaint/src/color.rs [hunk 1]
```diff
@@ -186,6 +186,36 @@ impl Color32 {
     }
 }
 
+#[macro_export]
```
Comment (emilk) [inline epaint/src/color.rs:189]:
```suggestion
/// Construct a [`Color32`] from a hex RGB or RGBA string.
///
/// ```
/// assert_eq!(hex_color!("#202122"), Color32::from_rgb(0x20, 0x21, 0x22));
/// assert_eq!(hex_color!("#abcdef12"), Color32::from_rgba_unmultiplied(0xab, 0xcd, 0xef, 0x12));
/// ```
#[macro_export]
```

**Review by emilk — CHANGES_REQUESTED** [review]
It's a bit of a shame that it is not `const` but yeah, not much to do about that right now. I think this looks good, but the macro needs a docstring!

File: epaint/src/color.rs (hunk 1 above)
Comment (newcomb-luke) [reply epaint/src/color.rs:195]: Done!

**Comment by coderedart** [comment]
is hex string assumed to be in linear space? or sRGB?
docs could mention that so its not confusing. 
`Color32` mention that `Internally this uses 0-255 gamma space sRGBA color with premultiplied alpha. Alpha channel is in linear space.`.

**Pull request merged**